src/view/components/EditLoanEntry/index.tsx

```diff
@@ -0,0 +1,191 @@
+}
+
+/**
+ * Button color
+ */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+/**
+ * Page for EditLoanEntry
+ */
+class EditLoanEntry extends React.Component<PropsType, OwnStateType> {
+
+  constructor(props: PropsType) {
+    super(props);
+    this.state = {
+      amountOfLoan: this.props.loanEntry.amount,
+    };
+  }
+
+  /** Create page buttons */
+  private renderCancelButton = () => this.renderButton('Cancel', 'INFO', this.onCancelPress);
+  private renderSaveButton = () => this.renderButton('Save', 'PRIMARY', this.onSavePress);
+
+  /** Handle pressing cancel button */
+  private onCancelPress = () => this.props.goBack();
+
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    let newEntry: ThunkUpdateRow<LoanEntry> = {
+      uuid: this.props.loanEntry.uuid,
+      toPersonUuid: this.props.loanEntry.toPersonUuid,
+      fromPersonUuid: this.props.loanEntry.fromPersonUuid,
+      amount: this.state.amountOfLoan,
+      currency: 'UGX',
+    };
+    this.props.updateLoanEntry(newEntry);
+    this.props.navigate(Route.FARMER);
+  }
+
+  /**
+   * Handle entry changes, update local state
+   */
+  private onAmountChange = (newAmount: string) => this.setState(state => ({ amountOfLoan: parseFloat(newAmount) }));
+
+  /**
+   * Returns a button with text, color, and onPress callback specified
+   */
+  private renderButton(text: string, color: ButtonColor, onPress: any) {
+    const isInfo = color === 'INFO';
+    const isPrimary = color === 'PRIMARY';
+
+    return (
+      <Col style={Styles.button}>
+        <Button block info={isInfo} primary={isPrimary} onPress={onPress}>
+          <Text>{text}</Text>
+        </Button>
+      </Col>
+    );
+  }
+
+  private renderHeader() {
+    return (
+      <Grid>
+        <Row style={Styles.headerRow}>
+          <H1>
+            {this.props.farmer.firstName} {this.props.farmer.lastName}
+          </H1>
+        </Row>
+        <Row style={Styles.headerRow}>
+          <Text style={Styles.header}>
+            {moment(this.props.loanEntry.datetime, 'ddd MMM DD Y kk:mm:ss ZZ').local().format('MMMM Do YYYY, h:mm:ss a')}
+          </Text>
+        </Row>
+      </Grid>
+    );
+  }
+
+  private formatEditRow(label: string, value: number | string, onChangeText: any) {
+    return (
+      <Grid>
+        <Row>
+          <Col>
+            <Text>{label}</Text>
+          </Col>
+          <Col>
+          <Item>
+            <Input onChangeText={onChangeText} keyboardType={'numeric'}>
+              <Text>{value}</Text>
+            </Input>
+          </Item>
+          </Col>
+        </Row>
+      </Grid>
+    );
+  }
+
+  private renderEditFields() {
+    return (
+      <View style={Styles.editView}>
+        {this.formatEditRow('Amount (UGX)', this.props.loanEntry.amount, this.onAmountChange)}
+      </View>
+    );
+  }
+
+  /** Render method for EditLoanEntry */
+  public render() {
+    return(
+      <Content padder style={{ backgroundColor: 'white' }}>
```

Comment:
The color styling should be coming from the `Styles` file, I think I need ti fix that for my files too.

Reply:
Updated